src/hierarchies/updaters/metropolis_updater.h

```diff
@@ -4,11 +4,40 @@
 #include "abstract_updater.h"
 #include "target_lpdf_unconstrained.h"
 
+//! Base class for updaters using a Metropolis-Hastings algorithm
+//!
+//! This class serves as the base for a CRTP.
+//! Children of this class should implement the methods
+//!     template <typename F>
+//!     Eigen::VectorXd sample_proposal(Eigen::VectorXd curr_state,
+//!                                     AbstractLikelihood &like,
+//!                                     AbstractPriorModel &prior, F
+//!                                     &target_lpdf)
+//! and
+//!     template <typename F>
+//!     double proposal_lpdf(Eigen::VectorXd prop_state, Eigen::VectorXd
+//!     curr_state,
+//!                          AbstractLikelihood &like, AbstractPriorModel
+//!                          &prior,
+//!                           F &target_lpdf)
```

Comment:
This indentation is kinda wonky, maybe move the second `Eigen::VectorXd` to the same line as `curr_state` and align it to the `(`? Same for `AbstractPriorModel`